Nifti: Added unit tests and a regression test for nifti test data

ResearchDaniel opened the pull request:
- New test data (they are quite small)
- Fix for data sets with cal_min/cal_max - ignore them for now
- New tests for ensuring that data is loaded and displayed correctly

@petersteneteg I had to change to PUCLIC in order for the nifti include directory to propagate to the unittest. Any idea if this should be done differently?
target_link_libraries(inviwo-module-nifti PUBLIC inviwo::niftiio inviwo::znz)

ResearchDaniel committed:
Nifti: Added unit tests and a regression test for nifti test data

- New test data (they are quite small)
- Fix for data sets with cal_min/cal_max - ignore them for now
- New tests for ensuring that data is loaded and displayed correctly

petersteneteg commented on modules/nifti/tests/unittests/nifti-test.cpp:
```diff
@@ -0,0 +1,77 @@
+using namespace inviwo;
+
+TEST(Nifti1, avg152T1_LR) {
+    // Stored in radiological convention
```
link?

ResearchDaniel replied:
Added more information about the data sets

petersteneteg commented on modules/nifti/tests/unittests/nifti-test.cpp:
```diff
@@ -0,0 +1,77 @@
+TEST(Nifti1, avg152T1_LR) {
+    // Stored in radiological convention
+    const auto filename =
+        filesystem::getPath(PathType::Tests, "/volumes/nifti/avg152T1_LR_nifti.nii.gz");
```
use the modules test dir instead ModulePath::TestVolumes

petersteneteg commented on modules/nifti/tests/unittests/nifti-test.cpp:
```diff
@@ -0,0 +1,77 @@
+    ASSERT_EQ(dvec2(0, 255), vol->dataMap_.dataRange);
+    ASSERT_EQ(dvec2(0, 255), vol->dataMap_.valueRange);
+    
+    ASSERT_EQ(mat3(vec3(-182, 0, 0), vec3(0, 218, 0), vec3(0, 0, 182)), vol->getBasis());
```
A left handed system?

ResearchDaniel replied:
Good point. It was flipped in the world transform. Changed so that we get a right-handed one.

petersteneteg commented on modules/nifti/src/niftireader.cpp:
```diff
@@ -209,6 +211,9 @@ std::shared_ptr<NiftiReader::VolumeSequence> NiftiReader::readData(const std::st
     }
 
     /*
+     We currently have no good way of considering cal_min and cal_max.
```
why can we not just use dataRange for this?

ResearchDaniel replied:
I managed to figure out how it could be done.

petersteneteg commented on modules/nifti/CMakeLists.txt:
```diff
@@ -14,13 +14,21 @@ set(SOURCE_FILES
 if(NOT IVW_USE_EXTERNAL_NIFTI)
 	add_subdirectory(ext/nifticlib-2.0.0)
-	target_link_libraries(inviwo-module-nifti PRIVATE inviwo::niftiio inviwo::znz)
+	target_link_libraries(inviwo-module-nifti PUBLIC inviwo::niftiio inviwo::znz)
```
Why does this have to be public? Are any nifty types exposed in the nifti module headers? The unittests should only need to link to the Inviwo module

ResearchDaniel committed:
Nifti: Review fixes

- Left-handed coordinate system -> Right-handed for data stored using radiological convention
- Correct value mapping with cal_min/cal_max (display range) is specified
- Moved nifti includes to cpp file.
- Moved test files to nifti module directory
- Updated regression test to reflect the poor color mapping of zstat1 due to cal_min/cal_max

ResearchDaniel commented:
Fixed all review comments

Inviwo-Team committed:
Jenkins: Format fixes

ResearchDaniel committed:
Nifti: Include fix

ResearchDaniel committed:
Copyright check: Skip zipped files (.gz)
